### PR TITLE
ilPluginAdmin, additional functions

### PR DESCRIPTION
**Pull request by nhaagen**
this is an improvement to ilPluginAdmin; 
most functions there require some knowledge about the plugin allready.
These function require no parameters, or at most the plugins's id.

They can be used to check on the availability of some plugin and ease the way of accessing it.

-getAllPlugins()
-getActivePlugins()
-isPluginActive($id)
-getPluginObjectById($id)

Here is the [short view](https://github.com/ILIAS-eLearning/ILIAS/pull/570/files?w=1).

### CODE REVIEW

**Comment by alex40724** [comment]
Looks good to me.

File: Services/Component/classes/class.ilPluginAdmin.php [hunk 1]
```diff
@@ -357,10 +357,74 @@ function hasLearningProgress($a_ctype, $a_cname, $a_slot_id, $a_pname)
...
+	 * Check, if a plugin is active
+	 *
+	 * @param 	string 	$id 	id of the plugin
+	 * @return 	boolean
+	 */
+	public static function isPluginActive($id) {
+		assert('is_string($id)');
+		$cached_component = ilCachedComponentData::getInstance();
+		$plugs = $cached_component->getIlPluginById();
+		if(array_key_exists($id, $plugs) &&	$plugs[$id]['active']) {
```
Comment (klees) [inline Services/Component/classes/class.ilPluginAdmin.php:400]: Replace tab by space at `&&    $plugs`.

**Review by klees — COMMENTED** [review]
LGTM.

**Review by klees — APPROVED** [review]

**Comment by chfsx** [comment]
Thanks @nhaagen, this is very helpful!

**Pull request merged**